src/main/java/duke/UI.java

```diff
@@ -0,0 +1,95 @@
+package duke;
+
+import java.util.function.Function;
+
+/**
+ * Class to handle User Interface interactions.
+ */
+public class UI {
+
+    Function<String, String> preprocessor;
+
+    /**
+     * Constructor for  UI with a custom preprocessor.
+     *
+     * @param preprocessor Function\<String,String\> to transform the input strings for display.
+     */
+    UI(Function<String, String> preprocessor) {
+        this.preprocessor = preprocessor;
+    }
+
+    /**
+     * Constructor for UI using default preprocessor.
+     */
+    UI() {
+        this(
+                (String source) -> {
+                    String border = "--------------------------------------------------";
+                    StringBuilder out = new StringBuilder();
+                    for (String line : source.split("\n")) {
+                        out.append("\t" + line + "\n");
+                    }
+                    return border + "\n"
+                            + out.toString()
+                            + border;
+                });
+    }
+
+    /**
+     * Displays the ascii startup logo.
+     */
+    public void displayLogo() {
+        String logo =
+                "  _____          _   _    ___   ___   ___   ___  \n" +
+                " |  __ \\   /\\   | \\ | |  / _ \\ / _ \\ / _ \\ / _ \\ \n" +
+                " | |  | | /  \\  |  \\| | | (_) | | | | | | | | | |\n" +
+                " | |  | |/ /\\ \\ | . ` |  \\__, | | | | | | | | | |\n" +
+                " | |__| / ____ \\| |\\  |    / /| |_| | |_| | |_| |\n" +
+                " |_____/_/    \\_\\_| \\_|   /_/  \\___/ \\___/ \\___/ \n";
+        System.out.println(logo);
+    }
+
+    /**
+     * Displays the welcome message.
+     */
+    public void displayWelcome() {
+        this.display("Hello, I'm DAN 9000\n"
+                + "How can I help you?");
+    }
+
+    /**
+     * Displays the provided string using preprocessor.
+     *
+     * @param content message String.
```

Comment:
Noticed the same capitalization issue here! Perhaps you could check your other javadocs for this issue as well?